titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/client/DefaultDirectKubeApiServerIntegrator.java

```diff
@@ -172,7 +169,7 @@ public boolean isReadyForScheduling() {
             Stopwatch timer = Stopwatch.createStarted();
             try {
                 logger.info("Deleting pod: {}", taskId);
-                kubeApiFacade.deleteNamespacedPod(KUBERNETES_NAMESPACE, taskId, DELETE_GRACE_PERIOD_SECONDS);
+                kubeApiFacade.deleteNamespacedPod(KUBERNETES_NAMESPACE, taskId, configuration.getDeleteGracePeriodSeconds());
```

Comment:
I think @sargun wants this to be a little more than the actual pod DeleteGracePeriodSeconds, to give time for the surrounding systems to react.

But IMO I think having them match (==) is the most correct?

Reply:
no strong opinion either way, but leaving the pod on apiserver a little longer than `DeleteGracePeriodSeconds` makes sense given there are propagation delays across multiple components beyond the kubelet/executor sleeping on `DeleteGracePeriodSeconds`